### PR TITLE
[Test/C-API] Add Tizen CPI test for GPU Delegate @open sesame 02/16 19:10

### PR DESCRIPTION
**Pull request by again4you**
This patch newly adds the test cases for TFLite GPU Delegate. It
includes both Single Shot APIs and Pipeline options.

Signed-off-by: Sangjung Woo <sangjung.woo@samsung.com>

**Self evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

### Known issue
* If **libOpenCL.so** library is not installed, TfLiteGpuDelegate Init function is failed. 
_ml_check_nnfw_availability (ML_NNFW_TYPE_TENSORFLOW_LITE, ML_NNFW_HW_GPU)_ does not check this problem.

### CODE REVIEW

File: tests/tizen_capi/unittest_tizen_capi.cc [hunk 1]
```diff
@@ -6651,6 +6651,125 @@ TEST (nnstreamer_capi_singleshot, invoke_dynamic_fail_n)
...
+
+  status = ml_tensors_data_create (in_info, &input);
+  EXPECT_EQ (status, ML_ERROR_NONE);
+  EXPECT_TRUE (input != NULL);
+
+  status = ml_single_invoke (single, input, &output);
+  EXPECT_EQ (status, ML_ERROR_NONE);
+  EXPECT_TRUE (output != NULL);
+
+  status = ml_tensors_data_get_tensor_data (output, 0, (void **)&data, &data_size);\
```
Comment (gichan-jang) [inline tests/tizen_capi/unittest_tizen_capi.cc:6709]:
```suggestion
  status = ml_tensors_data_get_tensor_data (output, 0, (void **)&data, &data_size);
```

File: tests/nnstreamer_filter_tensorflow2_lite/runTest.sh [hunk 2]
```diff
@@ -69,6 +69,7 @@ fi
 PATH_TO_MODEL="../test_models/models/mobilenet_v1_1.0_224_quant.tflite"
 PATH_TO_LABEL="../test_models/labels/labels.txt"
 PATH_TO_IMAGE="../test_models/data/orange.png"
+PATH_TO_OPENCL="/usr/lib/libOpenCL.so"
```
Comment (dongju-chae) [inline tests/nnstreamer_filter_tensorflow2_lite/runTest.sh:72]: I think it possibly causes false negative cases when checking the opencl installation.
For example, my desktop has opencl library located in `/usr/lib/x86_64-linux-gnu/libOpenCL.so`

File: tests/nnstreamer_filter_tensorflow2_lite/runTest.sh [hunk 3]
```diff
@@ -189,6 +190,26 @@ run_pipeline true:${auto_accl},cpu
 cat info | grep "accl = ${auto_accl}$"
 testResult $? 2-17 "accelerator set test" 0 1
 
+# GPU Delegate test
+if [ -e $PATH_TO_OPENCL]; then
```
Comment (dongju-chae) [inline tests/nnstreamer_filter_tensorflow2_lite/runTest.sh:194]: Need a space before `]`

File: tests/nnstreamer_filter_tensorflow2_lite/runTest.sh (hunk 2 above)
Comment (dongju-chae) [reply tests/nnstreamer_filter_tensorflow2_lite/runTest.sh:72]: How about using an environment variable set from a build script or other ways.

File: tests/nnstreamer_filter_tensorflow2_lite/runTest.sh (hunk 3 above)
Comment (again4you) [reply tests/nnstreamer_filter_tensorflow2_lite/runTest.sh:194]: ACK!

File: tests/tizen_capi/unittest_tizen_capi.cc (hunk 1 above)
Comment (again4you) [reply tests/tizen_capi/unittest_tizen_capi.cc:6709]: ACK!

File: tests/nnstreamer_filter_tensorflow2_lite/runTest.sh (hunk 2 above)
Comment (wooksong) [reply tests/nnstreamer_filter_tensorflow2_lite/runTest.sh:72]:
```
$ ldconfig -p | grep "libOpenCL.so " | awk '{print $4}'
/usr/local/cuda-10.2/targets/x86_64-linux/lib/libOpenCL.so
```

I am not sure that this works on Tizen.

File: tests/tizen_capi/unittest_tizen_capi.cc [hunk 4]
```diff
@@ -6651,6 +6651,125 @@ TEST (nnstreamer_capi_singleshot, invoke_dynamic_fail_n)
...
+  /* supposed to run test in build directory */
+  const gchar *root_path = g_getenv ("NNSTREAMER_SOURCE_ROOT_PATH");
+  if (root_path == NULL)
+    root_path = "..";
+
+  test_model = g_build_filename (root_path, "tests", "test_models", "models",
+      "mobilenet_v1_1.0_224_quant.tflite", NULL);
+  ASSERT_TRUE (g_file_test (test_model, G_FILE_TEST_EXISTS));
+
+  if (!g_file_test (opencl_path, G_FILE_TEST_EXISTS)) {
```
Comment (wooksong) [inline tests/tizen_capi/unittest_tizen_capi.cc:6745]: ```dlopen()``` without the absolute path tries to find the given shared object from LD_LIBRARY_PATH.

**Comment by again4you** [comment]
The API code is moved to another repo so I will close this old PR.

**Pull request closed without merging**